packages/common/src/spokenForms/modifiers.ts

```diff
@@ -0,0 +1,202 @@
+  selector: "selector",
+  statement: "state",
+  string: "string",
+  branch: "branch",
+  type: "type",
+  value: "value",
+  condition: "condition",
+  unit: "unit",
+  //  XML, JSX
+  xmlElement: "element",
+  xmlBothTags: "tags",
+  xmlStartTag: "start tag",
+  xmlEndTag: "end tag",
+  // LaTeX
+  part: "part",
+  chapter: "chapter",
+  subSection: "subsection",
+  subSubSection: "subsubsection",
+  namedParagraph: "paragraph",
+  subParagraph: "subparagraph",
+  environment: "environment",
+  // Text-based scope types
+  character: "char",
+  word: "word",
+  token: "token",
+  identifier: "identifier",
+  line: "line",
+  sentence: "sentence",
+  paragraph: "block",
+  document: "file",
+  nonWhitespaceSequence: "paint",
+  boundedNonWhitespaceSequence: "short paint",
+  url: "link",
+  notebookCell: "cell",
+
+  switchStatementSubject: null,
+};
+
+const surroundingPairsSpoken: Record<SurroundingPairName, string | null> = {
+  curlyBrackets: "curly",
+  angleBrackets: "diamond",
+  escapedDoubleQuotes: "escaped quad",
+  escapedSingleQuotes: "escaped twin",
+  escapedParentheses: "escaped round",
+  escapedSquareBrackets: "escaped box",
+  doubleQuotes: "quad",
+  parentheses: "round",
+  backtickQuotes: "skis",
+  squareBrackets: "box",
+  singleQuotes: "twin",
+  any: "pair",
+  string: "string",
+
+  // Used internally by the "item" scope type
+  collectionBoundary: null,
+
+  // This one can't be used as a scope type; only for wrapping
+  // whitespace: "void",
+};
+
+const surroundingPairsDelimiters: Record<
+  SurroundingPairName,
+  [string, string] | null
+> = {
+  curlyBrackets: ["{", "}"],
+  angleBrackets: ["<", ">"],
+  escapedDoubleQuotes: ['\\"', '\\"'],
+  escapedSingleQuotes: ["\\'", "\\'"],
+  escapedParentheses: ["\\(", "\\)"],
+  escapedSquareBrackets: ["\\[", "\\]"],
+  doubleQuotes: ['"', '"'],
+  parentheses: ["(", ")"],
+  backtickQuotes: ["`", "`"],
+  squareBrackets: ["[", "]"],
+  singleQuotes: ["'", "'"],
+
+  any: null,
+  string: null,
+  collectionBoundary: null,
+};
+const surroundingPairDelimiterToName = new CompositeKeyMap<
+  [string, string],
+  SurroundingPairName
+>((pair) => pair);
+
+for (const [name, pair] of Object.entries(surroundingPairsDelimiters)) {
+  if (pair != null) {
+    surroundingPairDelimiterToName.set(pair, name as SurroundingPairName);
+  }
+}
+
+export const surroundingPairForceDirections = {
+  left: "left",
+  right: "right",
+};
+
+export function modifierTypeToSpokenForm(modifier: ModifierType): string {
+  const result = modifiers[modifier];
+  if (result == null) {
+    throw Error(`Unknown modifier '${modifier}'`);
+  }
+  return result;
+}
+
+export function scopeTypeTypeToSpokenForm(scope: SimpleScopeTypeType): string {
+  if (scope === "switchStatementSubject") {
+    throw new NoSpokenFormError(`No spoken form for scope '${scope}'`);
+  }
+
+  const result = scopes[scope];
+  if (result == null) {
+    throw Error(`Unknown scope '${scope}'`);
+  }
+  return result;
+}
```

Comment:
It's a bit unfortunate that we can't distinguish between whether something has no spoken form or is an error just from the map. I'd be tempted to have something a bit stronger typed in our map, but I think we can revisit this when we start to actually get the spoken form from the user's customizations as per my other PR. In my PR we represent spoken forms as a list, and always have an entry for everything. Things that don't have a spoken form just have an empty list. More verbose and annoying here were we just want our one canonical default. But we may possibly use these to generate something closer to my format. We could also use a sentinel value to indicate there is no default spoken form, to distinguish from the other case where we generate the spoken form via other mechanisms (eg ordinal scope)

Reply:
Sounds reasonable to revisit this

Reply:
On second thought, I don't see why `switchStatementSubject` is any different from the section level scope types. They're not activated by default, but we still have them mapped in the spoken forms here. So why don't we just do the same thing with `switchStatementSubject`?

Reply:
`switchStatementSubject` has no default spoken form in our python files and does not exist in the users csv.

Reply:
sure but I'm not seeing why we shouldn't have a spoken form here?

Reply:
I would be inclined to have a spoken form for everything where it makes any sense to have a spoken form, as you've done for section headings

Reply:
if no objections i'll do that

Reply:
Just because we don't have one and I didn't think making up new spoken forms was part of this pr, but if you have something please add it. You also need to add it Talon side or our spoken form tests will fail.

Reply:
Let's leave it and revisit when we start supporting custom spoken forms